### PR TITLE
Extend Publisher and Consumer with application headers

### PR DESCRIPTION
**Pull request by alexanderM91**
- Developer(s): @alexanderM91

- Ticket: https://spryker.atlassian.net/browse/TE-549

- Academy PR: ACADEMY_URL_HERE

- Release Group: https://release.spryker.com/release-groups/view/1058

- Release Overview: https://release.spryker.com/core/TE-549 [Replace with actual PR overview link]

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   RabbitMq                 | minor                 |  Transfer                     |

#### Release Notes

Introduced possibility to support headers for AMQP messages.

-----------------------------------------

##### Change log

Improvements

- Adjusted `RabbitMqAdapter::receiveMessage()` to enable support of application headers.
- Adjusted `RabbitMqAdapter::receiveMessages()` to enable support of application headers.
- Adjusted `RabbitMqAdapter::sendMessage()` to enable support of application headers..
- Adjusted `RabbitMqAdapter::sendMessages()` to enable support of application headers.

### CODE REVIEW

File: src/Spryker/Client/RabbitMq/Model/Publisher/Publisher.php [hunk 1]
```diff
@@ -143,7 +144,17 @@ protected function publish(AMQPMessage $message, $exchangeQueue, $routingKey, $p
...
+        $headers = $messageTransfer->getHeaders();
+
+        if ($headers === null) {
+            return $message;
+        }
+
+        $headersTable = new AMQPTable($headers);
+        $message->set('application_headers', $headersTable);
+
+        return $message;
```
Comment (sokyrko) [inline src/Spryker/Client/RabbitMq/Model/Publisher/Publisher.php]:
```suggestion
         $message = new AMQPMessage($messageTransfer->getBody(), $this->config->getMessageConfig());
         $headers = $messageTransfer->getHeaders();
 
         if ($headers !== null) {
             $headersTable = new AMQPTable($headers);
             $message->set('application_headers', $headersTable);
          }
 
         return $message;
```

**Comment by sokyrko** [comment]
Please also adjust constraint table.

File: src/Spryker/Client/RabbitMq/Model/Consumer/Consumer.php [hunk 2]
```diff
@@ -128,6 +133,24 @@ public function collectQueueMessages(AMQPMessage $message)
...
 
+    /**
+     * @param \PhpAmqpLib\Message\AMQPMessage $message $message
+     * @param \Generated\Shared\Transfer\QueueSendMessageTransfer $queueSendMessageTransfer
+     *
+     * @return \Generated\Shared\Transfer\QueueSendMessageTransfer
+     */
+    protected function addApplicationHeaders(AMQPMessage $message, QueueSendMessageTransfer $queueSendMessageTransfer): QueueSendMessageTransfer
+    {
+        $messageProps = $message->get_properties();
```
Comment (stereomon) [inline src/Spryker/Client/RabbitMq/Model/Consumer/Consumer.php]: Please do not use abbreviations. Please use `$messageProperties`

Comment (alexanderM91) [reply src/Spryker/Client/RabbitMq/Model/Consumer/Consumer.php]: @stereomon, thank you, updated

File: composer.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "php": ">=7.1",
     "php-amqplib/php-amqplib": "^2.6",
     "spryker/kernel": "^3.20.0",
-    "spryker/queue": "^1.1.2",
+    "spryker/queue": "^1.3.0",
```
Comment (gerner-spryker) [inline composer.json]: After the changes were moved here, this increase is not necessary.

**Pull request merged**